pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -250,28 +297,49 @@ def cluster_environment(self) -> ClusterEnvironment:
         return self._cluster_environment
 
     @property
-    def on_cpu(self) -> bool:
-        return self._device_type == DeviceType.CPU
+    def has_cpu(self) -> bool:
+        return True
+
+    @property
+    def use_cpu(self) -> bool:
+        return self._accelerator_type == DeviceType.CPU
+
+    @property
+    def has_gpu(self) -> bool:
+        # Here, we are not checking for GPU availability, but instead if User has passed
+        # `gpus` to Trainer for training.
+        gpus = self.parallel_device_ids
+        return gpus is not None and len(gpus) > 0 and torch.cuda.is_available()
```

Comment:
```suggestion
        return gpus is not None and len(gpus) > 0
```
based on the description of this property, we would only check for the user input!

Reply:
Good point, done!